crates/autopilot/src/run_loop.rs

```diff
@@ -212,20 +261,103 @@ impl RunLoop {
                 return;
             }
 
-            tracing::info!(driver = %driver.name, "settling");
-            let submission_start = Instant::now();
-            match self
-                .settle(driver, solution, auction_id, block_deadline)
+            let solved_order_uids: HashSet<_> = solution.orders().keys().cloned().collect();
```

Comment:
Setting up all the stuff that is needed for the submission should probably be hidden away in the `settle()` function (or possibly some wrapper if that makes `settle()` to noisy) in order to keep `single_run()` as clean as possible.

Reply:
This is done to reduce memory footprint. Otherwise, the solution must be cloned entirely to be passed inside the async block where the settle function is called.

Reply:
Since most of the solutions settle very few orders (~1.4 on average) the `SolutionWithId` struct is also very small. Let's please first worry about readability/maintainability of these domain functions instead of memory footprint.

Reply:
Extracted the logic into a separate function for better readability of the `single_run` function.